pkg/healthz/healthz.go

```diff
@@ -10,21 +10,33 @@ import (
 )
 
 // NewHandler returns a new healthz handler.
-func NewHandler(p *plugin.V1Plugin) http.Handler {
-	return &handler{p: p}
+func NewHandler(p1s []*plugin.V1Plugin, p2s []*plugin.V2Plugin) http.Handler {
+	return &handler{p1s: p1s, p2s: p2s}
 }
 
 type handler struct {
-	p *plugin.V1Plugin
+	p1s []*plugin.V1Plugin
+	p2s []*plugin.V2Plugin
 }
 
 func (hd *handler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	err := hd.p.Health()
-	if err != nil {
-		rw.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprint(rw, err)
-		zap.L().Error("health check failed", zap.Error(err))
-		return
+	for _, p := range hd.p1s {
+		err := p.Health()
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprint(rw, err)
+			zap.L().Error("health check failed", zap.Error(err))
```

Comment:
will the naked err will tell us which plugin failed? if not please include the keyId in the error, like health check for plugin p.keyId failed or something

Reply:
Better yet we could poll each one and record the health of each one and emit an aggregate error.

but if you don't think this is going to be of any use in practice then ignore. I think in general we do want to know which of the 2+ plugins failed though.

Reply:
This will need more thought. We'll follow up in another PR if this proves to make debugging more difficult.